src/test/java/il/ac/bgu/cs/bp/bpjs/analysis/StateStoreTests.java

```diff
@@ -149,7 +151,7 @@ private void TestEqualJSVar(VisitedStateStore storeToUse) throws Exception {
         //now we want to compare specific
         BProgramSyncSnapshot state1 = snapshots.get(1).getSystemState();
         BProgramSyncSnapshot state2 = snapshots.get(2).getSystemState();
-        assertNotEquals(state1,state2);
+        assertEquals(state1, state2);
```

Comment:
Why was this changed from `assertNotEquals` to` assertEquals`?

Reply:
Note that in this function, i doesn't increment, so the state should be identical to the one 4 syncs back.

However, this is actually testing two things at once (snapshot and store) and I'll make it clearer what we're actually testing.